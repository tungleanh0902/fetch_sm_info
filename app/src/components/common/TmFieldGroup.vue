<template>
  <div class="tm-field-group">
    <slot />
  </div>
</template>

<script>
export default { name: `tm-field-group` }
</script>

<style>
.tm-field-group {
  display: flex;
  position: relative;
}

.tm-field-group .tm-field {
  flex: 1;
}

.tm-field-group .tm-field,
.tm-field-group .tm-btn-wrapper {
  margin: 0;
}

.tm-field-group .tm-btn-wrapper {
  max-width: 15rem;
}

.tm-field-group .tm-field + .tm-btn-wrapper {
  margin-left: 1rem;
}

.tm-field-group .tm-btn-wrapper + .tm-field {
  margin-left: 1rem;
}
</style>
