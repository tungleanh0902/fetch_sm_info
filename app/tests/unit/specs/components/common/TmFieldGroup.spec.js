import { mount } from "@vue/test-utils"
import TmFieldGroup from "common/TmFieldGroup"

describe(`TmFieldGroup`, () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(TmFieldGroup, {
      slots: { default: `<input class="fake-field" placeholder="Fake" />` },
    })
  })

  it(`has the expected html structure`, () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`has a slot with a field`, () => {
    expect(wrapper.findAll(`.fake-field`).length).toBe(1)
  })
})
