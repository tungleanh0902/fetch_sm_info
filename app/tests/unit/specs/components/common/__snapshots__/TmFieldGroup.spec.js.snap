// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmFieldGroup has the expected html structure 1`] = `
<div
  class="tm-field-group"
>
  <input
    class="fake-field"
    placeholder="Fake"
  />
</div>
`;
